pkg/upstream/cluster/cluster_manager.go

```diff
@@ -496,11 +506,15 @@ func (cm *clusterManager) getActiveConnectionPool(balancerContext types.LoadBala
 		if log.DefaultLogger.GetLogLevel() >= log.DEBUG {
 			log.DefaultLogger.Debugf("[upstream] [cluster manager] clusterSnapshot.loadbalancer.ChooseHost result is %s, cluster name = %s", addr, clusterSnapshot.ClusterInfo().Name())
 		}
-		value, ok := cm.protocolConnPool.Load(proto)
+		value, ok := cm.protocolConnPool[cm.protocolPoolIndex(clusterSnapshot)].Load(proto)
```

Comment:
There is some redundancy in these two steps.
`protocolConnPool` could be a new structure with his own functions, and two maps?

Reply:
`value, error := protocolConnPool.Get(cm, proto, snapshot)` 
how about this?

Reply:
Great idea.Maybe `shutdown` also can be modified in this way.